lib/datadog/tracing/sampling/matcher.rb

```diff
@@ -59,27 +105,10 @@ def tags_match?(trace)
             # can affect exact string matching (e.g. '400' matching '400.0').
             tag = format('%g', tag) if tag.is_a?(Numeric)
 
-            matcher === tag
+            matcher.match?(tag)
           end
         end
       end
-
-      # A {Datadog::Tracing::Sampling::Matcher} that allows for arbitrary trace matching
-      # based on the return value of a provided block.
-      class ProcMatcher < Matcher
```

Comment:
Arbitrary object matching is not required anymore, that this helper is no longer needed.